File: lib/widgets/market_summary_card.dart
import 'package:flutter/material.dart';
import '../models/asset.dart';
import '../utils/constants.dart';
import '../utils/formatters.dart';

class MarketSummaryCard extends StatelessWidget {
  final String market;
  final List<Asset> assets;

  const MarketSummaryCard({super.key, required this.market, required this.assets});

  @override
  Widget build(BuildContext context) {
    if (assets.isEmpty) return const SizedBox.shrink();

    final gainers = assets.where((a) => a.isGain).length;
    final losers = assets.length - gainers;
    final avgChange = assets.map((a) => a.change).reduce((a, b) => a + b) / assets.length;
    final totalVol = assets.map((a) => a.volume).reduce((a, b) => a + b);

    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        gradient: LinearGradient(
          colors: avgChange >= 0
              ? [AppColors.gainLight, AppColors.surface]
              : [AppColors.lossLight, AppColors.surface],
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
        ),
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: AppColors.border, width: 0.5),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Text(
                market,
                style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16, color: AppColors.textPrimary),
              ),
              const Spacer(),
              Container(
                padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
                decoration: BoxDecoration(
                  color: avgChange >= 0 ? AppColors.gainLight : AppColors.lossLight,
                  borderRadius: BorderRadius.circular(8),
                ),
                child: Text(
                  'Avg ${Formatters.change(avgChange)}',
                  style: TextStyle(
                    color: avgChange >= 0 ? AppColors.gain : AppColors.loss,
                    fontWeight: FontWeight.bold,
                    fontSize: 13,
                  ),
                ),
              ),
            ],
          ),
          const SizedBox(height: 12),
          Row(
            children: [
              _Metric(label: 'Gainers', value: '$gainers', color: AppColors.gain),
              const SizedBox(width: 24),
              _Metric(label: 'Losers', value: '$losers', color: AppColors.loss),
              const SizedBox(width: 24),
              _Metric(label: 'Volume 24H', value: Formatters.volume(totalVol), color: AppColors.accent),
            ],
          ),
        ],
      ),
    );
  }
}

class _Metric extends StatelessWidget {
  final String label;
  final String value;
  final Color color;
  const _Metric({required this.label, required this.value, required this.color});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(label, style: const TextStyle(fontSize: 11, color: AppColors.textSecondary)),
        Text(value, style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16, color: color)),
      ],
    );
  }
}

File: Views/Market/MarketSummaryCard.swift
import SwiftUI

struct MarketSummaryCard: View {
    let market: String
    let assets: [Asset]

    private var gainers:     Int    { assets.filter(\.isGain).count }
    private var losers:      Int    { assets.count - gainers }
    private var avgChange:   Double {
        guard !assets.isEmpty else { return 0 }
        return assets.map(\.change).reduce(0, +) / Double(assets.count)
    }
    private var totalVolume: Double { assets.map(\.volume).reduce(0, +) }

    var body: some View {
        if !assets.isEmpty {
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Text(market)
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.textPrimary)
                    Spacer()
                    Text("Avg \(Formatters.change(avgChange))")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(avgChange >= 0 ? .gain : .loss)
                        .padding(.horizontal, 10)
                        .padding(.vertical, 4)
                        .background(avgChange >= 0 ? Color.gainLight : Color.lossLight)
                        .cornerRadius(8)
                }
                HStack(spacing: 24) {
                    SummaryMetric(label: "Gainers",    value: "\(gainers)",                       color: .gain)
                    SummaryMetric(label: "Losers",     value: "\(losers)",                        color: .loss)
                    SummaryMetric(label: "Volume 24H", value: Formatters.volume(totalVolume),     color: .accent)
                }
            }
            .padding(16)
            .background(
                LinearGradient(
                    colors: avgChange >= 0
                        ? [Color.gainLight, Color.appSurface]
                        : [Color.lossLight, Color.appSurface],
                    startPoint: .topLeading,
                    endPoint:   .bottomTrailing
                )
            )
            .cornerRadius(16)
            .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color.appBorder, lineWidth: 0.5))
        }
    }
}

private struct SummaryMetric: View {
    let label: String
    let value: String
    let color: Color

    var body: some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(label)
                .font(.system(size: 11))
                .foregroundColor(.textSecondary)
            Text(value)
                .font(.system(size: 16, weight: .bold))
                .foregroundColor(color)
        }
    }
}
